### PR TITLE
Include documentation on how to make UI extensions work in airgap mode

### PR DESCRIPTION
**Pull request by martyav**
Fixes SURE-8077

## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, make sure to target the release branch instead of `main`.

## Description

Largely based on content from https://github.com/rancher/ui-plugin-charts?tab=readme-ov-file#air-gapped-rancher

Also reorganizes the air-gapped related content. Do you think maybe we should put it all on a separate page? Disadvantage is that some of other instructions are still relevant for air-gapped, so air-gapped users might not realize that deletion, uninstall, etc. are listed on another page.

### CODE REVIEW

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 1]
```diff
@@ -95,13 +98,104 @@ You must reload the page after disabling extensions or display issues may occur.
...
+
+Rancher provides some extensions, such as Kubewarden and Elemental, through the `ui-plugin-catalog` container image at https://hub.docker.com/r/rancher/ui-plugin-catalog/tags. If you're trying to install these extensions in an air-gapped environment, you must make the `ui-plugin-catalog` accessible.
+
+1. Mirror the `ui-plugin-catalog` image to a private registry:
+```bash
+export REGISTRY_ENDPOINT="my-private-registry.com"
+docker pull rancher/ui-plugin-catalog:1.0.0
+docker tag rancher/ui-plugin-catalog:1.0.0 $REGISTRY_ENDPOINT/rancher/ui-plugin-catalog:1.0.0
+docker push $REGISTRY_ENDPOINT/rancher/ui-plugin-catalog:1.0.0
+```
```
Comment (LucasSaintarbor) [inline docs/integrations-in-rancher/rancher-extensions.md:119]:
```suggestion

  ```bash
  export REGISTRY_ENDPOINT="my-private-registry.com"
  docker pull rancher/ui-plugin-catalog:1.0.0
  docker tag rancher/ui-plugin-catalog:1.0.0 $REGISTRY_ENDPOINT/rancher/ui-plugin-catalog:1.0.0
  docker push $REGISTRY_ENDPOINT/rancher/ui-plugin-catalog:1.0.0
  ```

```

Comment (LucasSaintarbor) [reply docs/integrations-in-rancher/rancher-extensions.md:119]: Updating formatting for this list of instructions:

Screenshot 2024-04-05 at 10 10 32 AM

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 2]
```diff
@@ -53,29 +53,32 @@ In v2.7.0, the built-in extensions aren't displayed under the **Available** tab.
...
+1. On the top right, click **⋮ > Manage Repositories**.
+1. Find the name of the extension repository you want to delete. Select the checkbox next to the repository name, then click **Delete**.
 
-1. Select the **Available** tab, and click the **Reload** button to make sure that the list of extensions is up to date.
+## Deleting Extension Repository Container Images
 
-1. Find the extension you just added, and click the **Install** button.
+1. Click **☰**, then select **Extensions**, under **Configuration**.
+1. On the top right, click **⋮ > Manage Extension Catalogs**.
+1. Find the name of the container image you want to delete. Click **⋮ > Uninstall**.
```
Comment (LucasSaintarbor) [inline docs/integrations-in-rancher/rancher-extensions.md:81]:
```suggestion
1. Find the name of the container image you want to delete, then click **⋮ > Uninstall**.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 3]
```diff
@@ -95,13 +98,104 @@ You must reload the page after disabling extensions or display issues may occur.
...
+  name: ui-plugin-catalog-repo
+spec:
+  url: http://ui-plugin-catalog-svc.cattle-ui-plugin-system:8080
+```
+
+After you successfully set up these resources, you can install the extensions from the `ui-plugin-charts` manifest into your air-gapped environment.
+
+### Importing and Installing Extensions in an Air-gapped Environment
+
+1. Find the address of the container image repository that you want to import as an extension. You should import and use the latest tagged version of the image to ensure you receive the latest features and security updates.
```
Comment (LucasSaintarbor) [inline docs/integrations-in-rancher/rancher-extensions.md:178]: Extra space between steps:

Screenshot 2024-04-05 at 10 21 23 AM

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 4]
```diff
@@ -95,13 +98,104 @@ You must reload the page after disabling extensions or display issues may occur.
...
+
+1. Click **☰**, then select **Extensions**, under **Configuration**.
+
+1. On the top right, click **⋮ > Manage Extension Catalogs**.
+
+1. Select the **Import Extension Catalog** button.
+
+1. Enter the image address in the **Catalog Image Reference** field. 
+
+    * **(Optional)** If the container image is private: Select the secret you just created from the **Pull Secrets** drop-down menu.
```
Comment (LucasSaintarbor) [inline docs/integrations-in-rancher/rancher-extensions.md:190]:
```suggestion
    * **(Optional)** If the container image is private, select the secret you just created from the **Pull Secrets** drop-down menu.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 5]
```diff
@@ -95,13 +98,104 @@ You must reload the page after disabling extensions or display issues may occur.
...
+
+1. Find the address of the container image repository that you want to import as an extension. You should import and use the latest tagged version of the image to ensure you receive the latest features and security updates.
+
+    * **(Optional)** If the container image is private: [Create](../how-to-guides/new-user-guides/kubernetes-resources-setup/secrets.md) a registry secret within the `cattle-ui-plugin-system` namespace. Enter the domain of the image address in the **Registry Domain Name** field.
+
+1. Click **☰**, then select **Extensions**, under **Configuration**.
+
+1. On the top right, click **⋮ > Manage Extension Catalogs**.
+
+1. Select the **Import Extension Catalog** button.
```
Comment (LucasSaintarbor) [inline docs/integrations-in-rancher/rancher-extensions.md:186]: Do we want to use the term button? Just noticed it wasn't used in the other instructions

Comment (martyav) [reply docs/integrations-in-rancher/rancher-extensions.md:186]: Good catch, that shouldn't be included.

**Review by btat — CHANGES_REQUESTED** [review]

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 6]
```diff
@@ -95,13 +98,94 @@ You must reload the page after disabling extensions or display issues may occur.
...
+
+Rancher provides some extensions, such as Kubewarden and Elemental, through the `ui-plugin-catalog` container image at https://hub.docker.com/r/rancher/ui-plugin-catalog/tags. If you're trying to install these extensions in an air-gapped environment, you must make the `ui-plugin-catalog` accessible.
+
+1. Mirror the `ui-plugin-catalog` image to a private registry:
+
+  ```bash
+  export REGISTRY_ENDPOINT="my-private-registry.com"
+  docker pull rancher/ui-plugin-catalog:1.0.0
+  docker tag rancher/ui-plugin-catalog:1.0.0 $REGISTRY_ENDPOINT/rancher/ui-plugin-catalog:1.0.0
+  docker push $REGISTRY_ENDPOINT/rancher/ui-plugin-catalog:1.0.0
```
[the remark this answers is not in the document]
Comment (martyav) [reply docs/integrations-in-rancher/rancher-extensions.md:119]: Thanks for noticing this. Missed it when I was moving the sections around.

File: docs/integrations-in-rancher/rancher-extensions.md (hunk 3 above)
Comment (martyav) [reply docs/integrations-in-rancher/rancher-extensions.md:178]: @LucasSaintarbor I think this should be fixed now

**Review by LucasSaintarbor — APPROVED** [review]

**Review by btat — APPROVED** [review]

**Comment by martyav** [comment]
Need to sync versions

**Comment by btat** [comment]
> Need to sync versions

Woops, sorry jumped the gun.

**Pull request merged**